src/jit/lclvars.cpp

```diff
@@ -495,8 +496,6 @@ void                Compiler::lvaInitRetBuffArg(InitVarDscInfo *    varDscInfo)
             }
         }
 
-        assert(genMapIntRegNumToRegArgNum(varDsc->lvArgReg) < MAX_REG_ARG);
```

Comment:
This assert is still holds except for Arm64.  Instead of deleting it should we add a new assert for ARM64?

e.g.

```
#ifdef _TARGET_AMD64_
   assert(genMapIntRegNumToRegArgNum(varDsc->lvArgReg) < MAX_REG_ARG ||
              (varDsc->lvArgReg == theFixedRetBuffReg()));
#else
   assert(genMapIntRegNumToRegArgNum(varDsc->lvArgReg) < MAX_REG_ARG);
#endif
```